code/modules/power/apc.dm

```diff
@@ -550,15 +550,14 @@ var/zapLimiter = 0
 	if (user.getStatusDuration("stunned") || user.getStatusDuration("weakened") || user.stat)
 		return
 
-	if ( (get_dist(src, user) > 1 ))
-		if (!issilicon(user) && !isAI(user))
-			src.remove_dialog(user)
-			user.Browse(null, "window=apc")
-			return
-		else if ((issilicon(user) || isAI(user)) && src.aidisabled)
-			boutput(user, "AI control for this APC interface has been disabled.")
-			user.Browse(null, "window=apc")
-			return
+	if (!in_range(src, user))
+		src.remove_dialog(user)
+		user.Browse(null, "window=apc")
+		return
+	else if (can_access_remotely(user) && src.aidisabled)
+		boutput(user, "AI control for this APC interface has been disabled.")
```

Comment:
While the underlying check here is comparable to the previous implementation, there's nothing requiring `can_access_remotely` to be driven by their AI/silicon status. As the "AI-disabled" bit is technically separate from whether it can be accessed remotely, I would look to leave as the `issilicon`/`isAI` checks for this.

Reply:
From a conceptual standpoint I viewed `can_access_remotely` as a "hey this can be controlled via packets, that argument seems supported in `receive_signal`. If the breadth of mobs that could perform actions remotely or if there was a device/augment that could help perform that action it would be easier to add that to `can_access_remotely`.